validate_test.go

```diff
@@ -83,8 +86,9 @@ func TestValidateUnregisteredRef(t *testing.T) {
 // TestValidateInvalidListenerNames tests that validateRegistrations fails on
 // invalid listener names.
 func TestValidateInvalidListenerNames(t *testing.T) {
-	type foo struct{}
+	type foo interface{}
 	type fooImpl struct {
+		_ Listener // undeclared name
```

Comment:
Mind updating the error list starting on line 109 to have an error to match this listener?